gopherage/cmd/junit/junit.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package junit
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/tools/cover"
+
+	"k8s.io/test-infra/gopherage/pkg/cov/junit"
+)
+
+type flags struct {
+	OutputFile string
+	Threshold  float32
+}
+
+// MakeCommand returns a `jnuit` command.
+func MakeCommand() *cobra.Command {
+	flags := &flags{}
+	cmd := &cobra.Command{
+		Use:   "junit [profile]",
+		Short: "Summarize coverage profile and produce the result in junit xml format.",
+		Long: `Summarize coverage profile and produce the result in junit xml format.
+Summary done at per-file and per-package level. Any coverage below coverage-threshold will be marked
+with a <failure> tag in the xml produced.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			run(flags, cmd, args)
+		},
+	}
+	cmd.Flags().StringVarP(&flags.OutputFile, "output", "o", "-", "output file")
+	cmd.Flags().Float32VarP(&flags.Threshold, "threshold", "t", .8, "code coverage threshold")
+	return cmd
+}
+
+func run(flags *flags, cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		fmt.Fprintln(os.Stderr, "expected exactly one argument: coverage file path")
+		os.Exit(1)
+	}
+
+	if flags.Threshold < 0 || flags.Threshold > 1 {
+		fmt.Fprintln(os.Stderr, "coverage threshold must be a float number between 0 to 1, inclusively")
+		os.Exit(1)
+	}
+
+	profilePath := args[0]
+
+	profiles, err := cover.ParseProfiles(profilePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to parse profile file: %v.", err)
+		os.Exit(1)
+	}
+
+	text, err := junit.ProfileToTestsuiteXML(profiles, flags.Threshold)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to produce xml from profiles: %v.", err)
+		os.Exit(1)
+	}
+
+	var file *os.File
+	if flags.OutputFile == "-" {
+		file = os.Stdout
+	} else {
+		file, err = os.Create(flags.OutputFile)
+
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to create file: %v.", err)
+			os.Exit(1)
+		}
+	}
+
+	if _, err = file.Write(text); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to produce xml from profiles: %v.", err)
```

Comment:
err msg should indicate "Failed to write xml to disk" or something like that. We already have the xml at this stage in text.